kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/steps/KafkaSteps.java

```diff
@@ -41,17 +50,19 @@ private KafkaSteps() {
      * @param replicas the replicas
      */
     public static void createTopic(String deployNamespace, String topicName, String bootstrap, int partitions, int replicas) {
-        String podName = Constants.KAFKA_ADMIN_CLIENT_LABEL + "-create";
-        kubeClient().getClient().run().inNamespace(deployNamespace).withNewRunConfig()
-                .withImage(Constants.TEST_CLIENTS_IMAGE)
-                .withName(podName)
-                .withRestartPolicy(NEVER_POLICY)
-                .withCommand(ADMIN_CLIENT_COMMAND)
-                .withArgs(TOPIC_COMMAND, "create", BOOTSTRAP_ARG + bootstrap, "--topic=" + topicName, "--topic-partitions=" + partitions,
-                        "--topic-rep-factor=" + replicas)
-                .done();
+        LOGGER.debug("Consuming messages from '{}' topic", topicName);
+        String name = Constants.KAFKA_ADMIN_CLIENT_LABEL + "-create";
+        List<String> args = Arrays.asList(TOPIC_COMMAND, "create", BOOTSTRAP_ARG + bootstrap, "--topic=" + topicName, "--topic-partitions=" + partitions,
+                "--topic-rep-factor=" + replicas);
+        InputStream file = KafkaUtils.replaceStringInResourceFile(ADMIN_CLIENT_TEMPLATE, Map.of(
+                NAMESPACE_VAR, deployNamespace,
+                NAME_VAR, name,
+                ARGS_VAR, String.join(DELIMITER, args),
+                KAFKA_VERSION_VAR, Environment.KAFKA_VERSION));
 
-        DeploymentUtils.waitForPodRunSucceeded(deployNamespace, podName, Duration.ofSeconds(30));
+        kubeClient().getClient().load(file).inNamespace(deployNamespace).create();
+        String podName = KafkaUtils.getPodNameByLabel(deployNamespace, "app", name, Duration.ofSeconds(30));
+        DeploymentUtils.waitForDeploymentRunning(deployNamespace, podName, Duration.ofSeconds(30));
```

Comment:
How is waiting for the pod to be running good enough?  Don't you need to let the pod terminate to know whether it is successful or not.